src/persist-client/src/internal/compact.rs

```diff
@@ -185,67 +193,80 @@ impl Compactor {
     {
         let parts_cpu_heavy_runtime = Arc::clone(&cpu_heavy_runtime);
         let compact_blocking = async move {
```

Comment:
can we pull this whole `async move {` block out into a method? I realize that pretty drastically guts the contents of `compact`, but I dunno, feels like the length has passed some critical threshold that's making it harder for me to follow (or, probably more likely, my original structure was bad and I only understood it because I wrote it :)